internal/access/credential.go

```diff
@@ -225,3 +234,43 @@ func checkPasswordRequirements(db data.ReadTxn, password string) error {
 
 	return nil
 }
+
+// checkPasswordBreached checks if the password is found in previous data breaches. It uses
+// Have I Been Pwned's API to perform this check. The API takes the first 5-characters prefix
+// of the SHA1 checksum of the password and returns a carriage returned list. A password is
+// breached if the remaining characters of its SHA1 checksum appears in this list.
+// Ref: https://haveibeenpwned.com/API/v3#PwnedPasswords
+func checkPasswordBreached(password string) error {
+	//nolint:gosec // this is required for the API
+	sum := sha1.Sum([]byte(password))
+	hexsum := hex.EncodeToString(sum[:])
+
+	request, err := http.NewRequestWithContext(context.Background(), http.MethodGet, fmt.Sprintf("https://api.pwnedpasswords.com/range/%s", hexsum[:5]), nil)
```

Comment:
I'm not really comfortable with the approach of calling a different service on account creation. I think we can just download the list from them and check it ourselves. I don't think this even needs to be updated all that frequently.

Also, like the password complexity stuff, this should be configurable per organization.

Reply:
I can see the concern with relying on an external service but I don't really see any reason for this to be an organization configuration.